apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -0,0 +1,55 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// log is for logging in this package.
+var log = logf.Log.WithName("baremetalhost-validation")
+
+// validateHost validates BareMetalHost resource for creation
+func (host *BareMetalHost) validateHost() []error {
+	log.Info("validate create", "name", host.Name)
+	var errs []error
+
+	if err := validateRAID(host.Spec.RAID); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errs
+}
+
+// validateChanges validates BareMetalHost resource on changes
+// but also covers the validations of creation
+func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
```

Comment:
Still on the naming side, I'd find more clear a name like `validateUpdate` for this method